### ISSUE TITLE
Reproducing paper results on Real Datasets

### ISSUE THREAD

**Issue by LuisOlCo** [sdv-dev/SDV#404]
### Environment details

If you are already running SDV, please indicate the following details about the environment in
which you are running it:

* SDV version: 0.9.0
* Python version: 3.6.1
* Operating System: Windows 10

### Problem description

Not an issue but a question. 
I am trying to reproduce the scores from the paper (Modeling Tabular Data using Conditional GAN)  for the real datasets, but I have not been able to get the same results.

While I was trying to reproduce these results, I used the default hyperparameters given in the class CTGANSynthesizer to train the CTGAN and to generate the synthetic data. Then I used the sdv.metrics.tabular to import the ML classifiers and to compute the F1 scores.

I am not expecting to obtain the exactly same results as the one reflected on table 5 of the paper, but the scores that I get are considerably lower, I even re-trained the model a few times but I kept getting kind of the same results, far from the ones given on the table.

I was wondering if these would be the steps for reproducing the same results, and is there any hyperparameters used on the paper that differs from the ones given by default?

Thank you very much!

### What I already tried

from ctgan import CTGANSynthesizer
import pandas as pd
from sklearn.model_selection import train_test_split
from dataset_preprocess import * # Contains Dataset class

# Load data
cur_dataset = Dataset('adult') # Dataset class loads the data file into pandas and gets the list of discrete columns
data_train, data_test= train_test_split(cur_dataset.df,test_size=cur_dataset.test_size,random_state=14)

# Fit train data into CTGAN model and save
ctgan = CTGANSynthesizer()
ctgan.fit(data_train,cur_dataset.discrete_columns,epochs=300)
ctgan.save('adult_synthetic_model_generator')

from sdv.demo import *
from sdv.tabular import CTGAN
from sdv.metrics.tabular import BinaryAdaBoostClassifier, BinaryLogisticRegression, BinaryDecisionTreeClassifier, MulticlassMLPClassifier

# Load CTGAN model and generate synthetic data
model = CTGAN.load('adult_synthetic_model_generator.pkl')
data_synthetic = model.sample(22700)

score_adaboost = BinaryAdaBoostClassifier.compute(data_synthetic,data_test,target='label')

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```

**Comment by npatki** [comment]
Hi @LuisOlCo, nice to meet you. What were the exact results you got?

I cannot speak to the exact steps that the paper uses, as I wasn't involved in creating it. One issue I see is that the `compute` method expects the real data as its first argument and then the synthetic data.

I would also suggest using the built-in SDV functionality for loading the data and modeling it. The `load_tabular_demo` functionality has these datasets; they are most likely to be the same ones used in the paper.

The code below worked for me:

```python
from sdv.demo import load_tabular_demo
from sdv.tabular import CTGAN

# load the dataset and metadata from the SDV demo library
metadata, data = load_tabular_demo('adult', metadata=True)

# create and fit the model
model = CTGAN(table_metadata=metadata)
model.fit(data)

# sample the same number of rows as the original data
synthetic_data = model.sample(num_rows=data.shape[0])

# evaluate the model
from sdv.metrics.tabular import BinaryAdaBoostClassifier, BinaryDecisionTreeClassifier, BinaryLogisticRegression, BinaryMLPClassifier

# input real data first, then synthetic data
print(BinaryAdaBoostClassifier.compute(data, synthetic_data, metadata=metadata, target='label'))
print(BinaryDecisionTreeClassifier.compute(data, synthetic_data, metadata=metadata, target='label'))
print(BinaryLogisticRegression.compute(data, synthetic_data, metadata=metadata, target='label'))
print(BinaryMLPClassifier.compute(data, synthetic_data, metadata=metadata, target='label'))
```

Printed scores were: 0.8987680567946698, 0.8385440265820873, 0.8297267162718729, 0.882487493723616.

This seems roughly in the same ballpark as the reported results from the `adult` dataset, which were within the 79-86% range.

**Comment by npatki** [comment]
Hello, this issue has been inactive for some time. I'll close it off since we've answered your initial question about reproducing the results. Feel free to reply if there is more to discuss and I'll reopen the issue.
